cmd/volume.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/mikelangelo-project/capstan/hypervisor"
+	"github.com/mikelangelo-project/capstan/hypervisor/qemu"
+)
+
+// Volume is an extended version of hypervisor.Volume with volume creation data.
+type Volume struct {
+	hypervisor.Volume
+	SizeMB int64
+	Name   string
+}
+
+const VOLUMES_DIR string = "volumes"
+
+// CreateVolume creates volume with given specifications.
+func CreateVolume(packagePath string, volume Volume) error {
+	if _, err := os.Stat(filepath.Join(packagePath, "meta", "package.yaml")); os.IsNotExist(err) {
+		return fmt.Errorf("Must be in package root directory")
+	}
+	volumesDir := filepath.Join(packagePath, VOLUMES_DIR)
+	if _, err := os.Stat(volumesDir); os.IsNotExist(err) {
+		if err := os.Mkdir(volumesDir, 0777); err != nil {
```

Comment:
Do we always create `777` directories?

Reply:
Updated to `744`. There has to be execute permission for `qemu-img` to be able to create the volume. I remember even 744 wasn't sufficient for unit tests when I was implementing this, but it seems to be somehow working now.